### DISCUSSION TITLE
finding documents in docket search?

### DISCUSSION THREAD

**Discussion by legaltextai** [freelawproject/courtlistener#3296] [Q&A]
Question from a newbie -- Is it possible to download petitions, briefs or any other documents , submitted by either party, other than court opinions? 
For example, I'd like to find habeas corpus petitions filed in a specific jurisdition. 
I see some relevant categories mentioned [here](https://github.com/freelawproject/courtlistener/blob/9591f6f2ee321584a7177f5397dd13a51a2d48a1/cl/recap/constants.py#L313)  What 's the best way to achieve that? 
Thanks

**Comment by mlissner** [comment]
You could do a search for the nature of suit in the front end, or you might be able to filter to it using one of the filters. There should be some good references to the nature of suit code elsewhere in this forum. If you can't find examples or if you can share what you've tried, that'd be helpful.

**Reply by legaltextai** [reply]
Tried [this](https://www.courtlistener.com/?q=&type=r&order_by=score%20desc&nature_of_suit=%22habeas%20corpus%22&filed_after=10%2F10%2F2023) search. With just "habeas corpus" in the nature of suit. Some documents are available, some are marked "Buy on Pacer". The latter I presume is the decision of the court? Will API query on recap archive yield the same result?

**Reply by legaltextai** [reply]
Seems like all recap apis require pacer credentials?

**Reply by mlissner** [reply]
Most documents cost money in PACER. The ones we have we get from the sources listed on our coverage page: 

https://www.courtlistener.com/help/coverage/

When we have the metadata for a document but nobody has contributed it to us yet, that's when it's still needed from PACER. 

We have a number of APIs, some of which require special access. We have one that requires PACER credentials (the "Fetch" API). If you want to discuss that, please send me an email with some details of the project you intend to do and we can get the ball rolling. My email is mike@free.law
